File: Data Structures & Algorithms/two-integer-sum/submission-3.swift
class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        var store = [Int: Int]()

        for (index, num) in nums.enumerated() {
            let diff = target - num

            if let prev = store[diff] {
                return [prev, index]
            }

            store[num] = index

        }

        return []
    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-3.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        unordered_map<int, int> mp;

        for (int i = 0; i < numbers.size(); i++) {
            int temp = target -  numbers[i];

            if (mp.count(temp)) {
                return { mp[temp], i + 1};
            }
            mp[numbers[i]] = i + 1;
        }

        return {};
    }
};
